airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -183,15 +183,28 @@ public AirbyteCatalog discover(JsonNode config) throws Exception {
 
   @Override
   public JsonNode toDatabaseConfig(JsonNode config) {
+    List<String> additionalParameters = new ArrayList<>();
+
     final StringBuilder jdbc_url = new StringBuilder(String.format("jdbc:mysql://%s:%s/%s",
         config.get("host").asText(),
         config.get("port").asText(),
         config.get("database").asText()));
+
     // see MySqlJdbcStreamingQueryConfiguration for more context on why useCursorFetch=true is needed.
     jdbc_url.append("?useCursorFetch=true");
     if (config.get("jdbc_url_params") != null && !config.get("jdbc_url_params").asText().isEmpty()) {
       jdbc_url.append("&").append(config.get("jdbc_url_params").asText());
     }
+
+    if (config.has("ssl") && config.get("ssl").asBoolean()) {
+      jdbc_url.append("&");
+      additionalParameters.add("useSSL=true");
+      additionalParameters.add("requireSSL=true");
+      additionalParameters.add("verifyServerCertificate=false");
+    }
+
+    additionalParameters.forEach(x -> jdbc_url.append(x).append("&"));
```

Comment:
Nitpick. This will result in an extra `&` at the end. You can use `String.join("&", additionalParameters)` to avoid it. Actually `additionalParameters` does not seem necessary. To follow the pattern of the previous `if` block, these parameters can just be appended to `jdbcUrl`:

```suggestion
    if (config.has("ssl") && config.get("ssl").asBoolean()) {
      jdbcUrl.append("&").append("useSSL=true")
          .append("&").append("requireSSL=true")
          .append("&").append("verifyServerCertificate=false")
    }
```

Reply:
Alternatively, the `additionalParameters` can be a class level static constant. So that it can be reused in unit test (`MySqlSSLSourceAcceptanceTest.java`).

```java
public static final List<String> SSL_PARAMETERS = List.of(
    "useSSL=true",
    "requireSSL=true",
    "verifyServerCertificate=false");
```

Then you can do

```java
if (config.has("ssl") && config.get("ssl").asBoolean()) {
  jdbcUrl.append("&").append(String.join("&", SSL_PARAMETERS));
}
```

Reply:
Updated, thanks!